### PR TITLE
Update `linkerd upgrade` instructions

### PR DESCRIPTION
**Pull request by alpeb**
Replace instructions to use `kubectl apply --prune` with `linkerd prune`

### CODE REVIEW

**Comment by alpeb** [comment]
`prune` is getting reverted for now (#10709), will add this later.

**Comment by alpeb** [comment]
Resurrecting, thanks to #10710

File: cli/cmd/upgrade.go [hunk 1]
```diff
@@ -79,16 +79,13 @@ A full list of configurable values can be found at https://www.github.com/linker
 `,
 
 		Example: `  # Upgrade CRDs first
-  linkerd upgrade --crds | kubectl apply --prune --prune-whitelist=apiextensions.k8s.io/v1/customresourcedefinitions
+  linkerd upgrade --crds | kubectl apply -f -
 
-  # Then upgrade the controle-plane and remove linkerd resources that no longer exist in the current version
-  linkerd upgrade | kubectl apply --prune -l linkerd.io/control-plane-ns=linkerd -f -
+  # Then upgrade the controle-plane
```
Comment (hawkw) [inline cli/cmd/upgrade.go:84]: typo:
```suggestion
  # Then upgrade the control plane
```

File: cli/cmd/upgrade.go [hunk 2]
```diff
@@ -79,16 +79,13 @@ A full list of configurable values can be found at https://www.github.com/linker
...
+  linkerd upgrade | kubectl apply -f -
 
-  # Then run this again to make sure that certain cluster-scoped resources are correctly pruned
-  linkerd upgrade | kubectl apply --prune -l linkerd.io/control-plane-ns=linkerd \
-  --prune-whitelist=rbac.authorization.k8s.io/v1/clusterrole \
-  --prune-whitelist=rbac.authorization.k8s.io/v1/clusterrolebinding \
-  --prune-whitelist=apiregistration.k8s.io/v1/apiservice -f -`,
+  # And lastly remove linkerd resources that no longer exist in the current version
```
Comment (hawkw) [inline cli/cmd/upgrade.go:87]: nit: how about
```suggestion
  # And lastly, remove linkerd resources that no longer exist in the current version
```

**Review by hawkw — APPROVED** [review]
couple tiny suggestions, but LGTM!

**Pull request merged**